src/classes/CSVReaderTest.cls

```diff
@@ -0,0 +1,151 @@
+/**
+* File Name :   CSVReaderTest.cls
+* Description : Made to only test the methods in CSVReader.cls
+*
+* Modification Log
+* ============================================================================
+* Ver Date       Author         Modification
+* --- ---------- -------------- --------------------------
+* 0.1 07/04/2014 Klea Kolaric   Initial version
+*
+*
+*/
+@isTest
+private class CSVReaderTest {
+
+    static testMethod void parseBasicCsvString()
+    {
+        Blob blobCSV = Blob.valueOf(basicCsvString);
+        
+        Integer numberOfRows = basicCsvString.split(CSVReader.lineTerminator).size();
+        Integer numberOfCols = basicCsvString.split(CSVReader.lineTerminator).get(0).split(',').size();
+        
+        Test.startTest();
+        List<List<String>> resultMatrix = CSVReader.readCSVFile(blobCSV);
+        Test.stopTest();
+        
+        system.assertEquals(resultMatrix.size(),numberOfRows);
+        for(List<String> rowList : resultMatrix)
+        {
+        	 system.assertEquals(rowList.size(),numberOfCols);
+        	 for(String strVal : rowList)
+        	 {
+        	 	system.assert(String.isNotBlank(strVal));
+        	 }
+        }
+    }
+    
+    static testMethod void parseQuotedCsvString()
```

Comment:
Style issue: test methods should be annotated with @IsTest for better readibility